crates/store/src/server/db_maintenance.rs

```diff
@@ -19,11 +19,14 @@ impl DbMaintenance {
         loop {
             tokio::time::sleep(self.optimization_interval).await;
 
-            info!(target: COMPONENT, "Starting database optimization");
+            let root_span = tracing::info_span!(
+                "optimize_database",
+                interval = self.optimization_interval.as_secs_f32()
+            );
 
-            match self.state.optimize_db().await {
-                Ok(_) => info!(target: COMPONENT, "Finished database optimization"),
-                Err(err) => error!(target: COMPONENT, %err, "Database optimization failed"),
+            {
+                let _enter = root_span.enter();
+                self.state.optimize_db().await.unwrap_or_else(|err| root_span.set_error(&err));
```

Comment:
Spans unfortunately don't work with async because the future may be moved to a separate thread of execution. https://docs.rs/tracing/latest/tracing/struct.Span.html#in-asynchronous-code

Instead one needs to instrument it.

```suggestion
            self.state.optimize_db().instrument(root_span).await.inspact_err(|err| root_span.set_error(&err));
```

Reply:
Good point, thank you! Unfortunately, suggested code can't be compiled because `root_span` is consumed by `instrument` method, but used later in the `inspect_err` method.

Reply:
They're cheap to clone iirc; or at least that's how I've been using them https://github.com/0xPolygonMiden/miden-node/blob/7546dc77036fcf8acf4a339cde5b5cb91051877e/crates/block-producer/src/block_builder/mod.rs#L103-L120